### PR TITLE
fix: showing last_synced_at for amex feed cards in dashboard

### PR DESCRIPTION
**Pull request by Aniruddha-Shriwant**
### <samp>🤖 Generated by Copilot at 331315d</samp>

This pull request adds support for American Express corporate cards to the Fyle app. It updates the `corporate-card.component` to show different logos and sync dates for different card types, and adds the `AMEX_FEED` enum value and mock data. It also adds and modifies unit tests and CSS styles for the new feature.

### <samp>🤖 Generated by Copilot at 331315d</samp>

> _Sing, O Muse, of the glorious deeds of Fyle,_
> _The agile app that tracks expenses with ease and style._
> _How they added to their `DataFeedSource` a new enum value,_
> _To integrate the cards of Amex, the mighty and the valuable._

### <samp>🤖 Generated by Copilot at 331315d</samp>

*  Add support for AMEX_FEED as a new data feed source for corporate cards ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-dd67b374788a52e6a98ad115d0a62ec5e63da0947b942e366160bec45f435e36R4), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-b5291bcd0d2b5a1b063451261863b38cf00ebc5077f21303f91b991408d84204R118-R140), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39L4-R21), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39L53-R71), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-89807115c69d39cdf0236632c05103783259617b4795b281ff94b44f05c95d2eR34-R40), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR8), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cL95-R107), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR188-R197), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR238-R253))
  *  Define a new enum value for AMEX_FEED in `data-feed-source.enum.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-dd67b374788a52e6a98ad115d0a62ec5e63da0947b942e366160bec45f435e36R4))
  *  Create a mock data object for an AMEX_FEED card in `platform-corporate-card.data.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-b5291bcd0d2b5a1b063451261863b38cf00ebc5077f21303f91b991408d84204R118-R140))
  *  Show a different logo for AMEX_FEED cards in the `corporate-card.component.html` template ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39L4-R21))
  *  Display the last synced date for AMEX_FEED cards in the `corporate-card.component.html` template ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-1c08c3139af341f98f5dc83524b76783d3ef70c489281b15d00d2da63c1c0f39L53-R71))
  *  Add CSS styles for the AMEX logo icon in `corporate-card.component.scss` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-89807115c69d39cdf0236632c05103783259617b4795b281ff94b44f05c95d2eR34-R40))
  *  Import the mock data object and the computeStackId function in `corporate-card.component.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR8), [link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR18))
  *  Test that the AMEX logo icon is shown for AMEX_FEED cards in `corporate-card.component.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cL95-R107))
  *  Test that the more options button is not visible for AMEX_FEED cards in `corporate-card.component.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR188-R197))
  *  Test that the last synced date is shown and formatted correctly for AMEX_FEED cards in `corporate-card.component.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2510/files?diff=unified&w=0#diff-393d21e7c8a24b907917f5b801dbd08fc2361a37bd2ea29d075aac4598fa406cR238-R253))

## Clickup
https://app.clickup.com/t/85zu4cup6

## Code Coverage
image

## UI Preview

Before:
image

After:

image

image

### CODE REVIEW

File: src/app/shared/components/corporate-card/corporate-card.component.html [hunk 1]
```diff
@@ -50,10 +61,14 @@
       </div>
     </ng-container>
 
-    <ng-container *ngIf="card.data_feed_source === dataFeedSourceTypes.YODLEE">
+    <ng-container
+      *ngIf="
+        card.data_feed_source === dataFeedSourceTypes.YODLEE || card.data_feed_source === dataFeedSourceTypes.AMEX_FEED
+      "
+    >
```
Comment (Aniruddha-Shriwant) [inline src/app/shared/components/corporate-card/corporate-card.component.html:68]: maybe linter did this weird linting changes

Comment (Aniruddha-Shriwant) [reply src/app/shared/components/corporate-card/corporate-card.component.html:68]: just or condition is added extra

File: src/assets/images/amex-logo.png
Comment (Chethan-Fyle) [inline src/assets/images/amex-logo.png:1]: 90KB is too much. Scale down the image to 200x200.

**Review by Chethan-Fyle — APPROVED** [review]

File: src/assets/images/amex-logo.png
Comment (Aniruddha-Shriwant) [reply src/assets/images/amex-logo.png:1]: done

**Pull request merged**